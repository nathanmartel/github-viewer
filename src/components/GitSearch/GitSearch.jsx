import React, { Component } from 'react';

export default class GitSearch extends Component {
  
  state = {
    inputText: ''
  }

  handleChange = ({ target }) => {
    this.setState({ [target.name]: target.value });
  };
  
  render() {
    const { inputText } = this.state;
    const { onSubmit } = this.props;

    return (
      <form onSubmit={onSubmit}>
        <label>GitHub username:
          <input type='text' onChange={this.handleChange} value={inputText}></input>
        </label>
        <button type='submit'>Search</button>
      </form>
    );
  }
}
