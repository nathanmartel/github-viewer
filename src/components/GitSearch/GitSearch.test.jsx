import React from 'react';
import { shallow } from 'enzyme';
import GitSearch from './GitSearch';

describe('GitSearch component', () => {
  it('renders GitSearch', () => {
    const wrapper = shallow(<GitSearch />);
    expect(wrapper).toMatchSnapshot();
  });
});
