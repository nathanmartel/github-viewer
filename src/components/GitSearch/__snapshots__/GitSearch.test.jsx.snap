// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GitSearch component renders GitSearch 1`] = `
<form>
  <label>
    GitHub username:
    <input
      onChange={[Function]}
      type="text"
      value=""
    />
  </label>
  <button
    type="submit"
  >
    Search
  </button>
</form>
`;
